### PR TITLE
feat(dsg): add static comments to generated base folder files

### LINKED ISSUE #1345: Data Service Generator: add to the base folder *dont overwrite*
in the base folder in the code that generated adding comment that say not to overwrite the contant to prevent git collision

### PR DESCRIPTION
**Pull request by michizhou**
Add static comments to all files in the Data Service Generator base folder to notify the developer to not change generated code manually and to expect any manual change to be overwritten, serving as a warning to users.
Fixes #1345

### CODE REVIEW

**Comment by yuval-hazaz** [comment]
@michizhou 
For this task, you need to add the comments only to the files in the base folder - mainly to :
- [entity]Service.base.ts 
- [entity]Module.base.ts 
- [entity]Resolver.base.ts 
- [entity]Controller.base.ts

but also, to all other DTOs and models in that folder.

in order to do that, I recommend creating a function in `packages/amplication-data-service-generator/src/util/ast.ts` and then using this function from all other locations.
It will require using AST - if you are not familiar with it, try to look on the other examples in the same file (or let us know and we can help)

something like this 
```typeScript
export function addAutoGenerationComment (
  file: namedTypes.File,
){

}
```


Your current change only impacts the generation test and not the actual generation.

I suggest that you start with the new function I mentioned above.
Then you should call the function from line 296 in `packages/amplication-data-service-generator/src/server/resource/service/create-service.ts` (in function `createServiceBaseModule`).

Once you complete this first location, we will guide you on how to move forward.

Let me know if you need anything to proceed. 

and thanks for contributing to the project!

File: packages/amplication-data-service-generator/scripts/generate-test-data-service.ts [hunk 1]
```diff
@@ -6,6 +6,11 @@ import appInfo from "../src/tests/appInfo";
 import { AppInfo, Module } from "../src/types";
 import { createDataService } from "../src/create-data-service";
 
+const STATIC_COMMENT =
+  "// **************************************************************************\n" +
+  "// GENERATED CODE - DO NOT MODIFY BY HAND\n" +
+  "// **************************************************************************\n";
+
 if (require.main === module) {
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/scripts/generate-test-data-service.ts:14]: you should not change this file, please see my comments on the PR.

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Comment by michizhou** [comment]
Hi @yuval-hazaz, I have modified my implementation of the static comments for the base folder files due in large part to your suggestions by writing a new function and calling it where appropriate. I'm not sure if my implementation of the function is totally correct given I was only able to add the static comment to the `file.comments`, so I'd appreciate it if you can provide feedback and guide me in the right direction if needed. Please also let me know if there are additional locations where I would need to call my function. Thank you for your time and consideration.

File: packages/amplication-data-service-generator/scripts/generate-test-data-service.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ async function writeModules(
     modules.map(async (module) => {
       const filePath = path.join(destination, module.path);
       await fs.promises.mkdir(path.dirname(filePath), { recursive: true });
-      await fs.promises.writeFile(filePath, module.code);
+      await fs.promises.writeFile(filePath, STATIC_COMMENT + module.code);
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/scripts/generate-test-data-service.ts:37]: Undo this change. 
The test files are generated based on the logic on DSG and should not be altered at this point

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
1. Please undo the changes in packages/amplication-data-service-generator/scripts/generate-test-data-service.ts

2. Please run the task "update-snapshot" in DSG. It will update a snapshot test file with the latest change.

image

File: packages/amplication-data-service-generator/scripts/generate-test-data-service.ts (hunk 2 above)
Comment (michizhou) [reply packages/amplication-data-service-generator/scripts/generate-test-data-service.ts:37]: @yuval-hazaz My deepest apologies for overlooking this. I'll be sure to correct the mistake with my latest changes.

Comment (michizhou) [reply packages/amplication-data-service-generator/scripts/generate-test-data-service.ts:37]: @yuval-hazaz My deepest apologies for overlooking this. I'll correct the mistake with my latest changes.

**Review by yuval-hazaz — DISMISSED** [review]

**Comment by yuval-hazaz** [comment]
@michizhou please note that some tests are failing
Please fix the relevant tests, or let me know if you need any guidance

**Comment by michizhou** [comment]
@yuval-hazaz I have updated the failing tests and verified that they do pass locally so there should be no problems with them now. Please let me know if any additional changes need to be made before merging.

**Review by yuval-hazaz — APPROVED** [review]

**Comment by yuval-hazaz** [comment]
@all-contributors please add @michizhou for code

**Comment by yuval-hazaz** [comment]
PR #1871 fixes this issue partially.
We leave this issue open for completing the task for all other auto-generated files
